### PR TITLE
[CIS-1445] Add test coverage for CIS-1339 / #1707

### PR DESCRIPTION
**Pull request by polqf**
### 🔗 Issue Link
https://stream-io.atlassian.net/browse/CIS-1445

### 🎯 Goal

This PR adds test coverage for #1707 

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change follows zero ⚠️ policy (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift [hunk 1]
```diff
@@ -1232,4 +1232,102 @@ class MessageDTO_Tests: XCTestCase {
...
+        let messages = [message, existingReplyNotInChannel, existingReplyInChannel, deletedReplyNotInChannel, deletedReplyInChannel]
+
+        try messages.forEach { messagePayload in
+            try database.writeSynchronously { session in
+                // Save the message
+                try session.saveMessage(payload: messagePayload, for: channelId, syncOwnReactions: true)
+            }
+        }
+
+
```
Comment (bielikb) [inline Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift:1313]:
```suggestion

```

File: Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift [hunk 2]
```diff
@@ -1232,4 +1232,102 @@ class MessageDTO_Tests: XCTestCase {
...
+            for: channelId.rawValue,
+            deletedMessagesVisibility: .visibleForCurrentUser,
+            shouldShowShadowedMessages: false
+        )
+
+        var retrievedMessages: [MessageDTO] = []
+        do {
+            retrievedMessages = try database.viewContext.fetch(request)
+        } catch {
+            XCTFail()
```
Comment (bielikb) [inline Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift:1326]:
```suggestion
            XCTFail(message: "Failed to fetch messages from database")
```

**Review by bielikb — APPROVED** [review]
Looks good. 

Here are just few comments

File: Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift [hunk 3]
```diff
@@ -1232,4 +1232,101 @@ class MessageDTO_Tests: XCTestCase {
...
+        )
+
+        let messages = [message, existingReplyNotInChannel, existingReplyInChannel, deletedReplyNotInChannel, deletedReplyInChannel]
+
+        try messages.forEach { messagePayload in
+            try database.writeSynchronously { session in
+                // Save the message
+                try session.saveMessage(payload: messagePayload, for: channelId, syncOwnReactions: true)
+            }
+        }
```
Comment (evsaev) [inline Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift:1311]: nit:
```suggestion
    try database.writeSynchronously { session in
        messages.forEach { messagePayload in
                // Save the message
                try session.saveMessage(payload: messagePayload, for: channelId, syncOwnReactions: true)
            }
        }
```

File: Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift [hunk 4]
```diff
@@ -1232,4 +1232,101 @@ class MessageDTO_Tests: XCTestCase {
...
+        var retrievedMessages: [MessageDTO] = []
+        do {
+            retrievedMessages = try database.viewContext.fetch(request)
+        } catch {
+            XCTFail("Failed to retrieve messages")
+        }
+
+        XCTAssertEqual(retrievedMessages.count, 2)
+        XCTAssertEqual(retrievedMessages.first?.id, messageId)
+        XCTAssertEqual(retrievedMessages.last?.id, existingReplyInChannelId)
```
Comment (evsaev) [inline Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift:1330]: nit:
```suggestion
        XCTAssertEqual(
            retrievedMessages.map(\.id),
            [messageId, existingReplyInChannelId]
        )
```

**Review by adamrushy — APPROVED** [review]
Awesome 🔥

File: CHANGELOG.md [hunk 5]
```diff
@@ -11,6 +11,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Fix multiple pagination requests being fired from `ChatChannelVC` and `ChatChannelListVC` [#1706](https://github.com/GetStream/stream-chat-swift/issues/1706)
 - Fix rendering unavailable reactions on `ChatMessageReactionAuthorsVC` [#1719](https://github.com/GetStream/stream-chat-swift/issues/1719)
 - Fix quoted messages not updated after edit [#1703](https://github.com/GetStream/stream-chat-swift/pull/1703)
+- Fix deleted replies being shown in channel [#1707](https://github.com/GetStream/stream-chat-swift/pull/1707)
```
Comment (evsaev) [inline CHANGELOG.md:14]: shouldn't we refer to the PR where this was actually fixed?

**Review by evsaev — APPROVED** [review]
Looks good, thanks 👍

**Pull request merged**